lib/puppet/pops/validation/checker4_0.rb

```diff
@@ -481,6 +482,18 @@ def check_ResourceTypeDefinition(o)
     internal_check_no_idem_last(o)
   end
 
+  def internal_check_parameter_type_literal(o)
+    type = nil
```

Comment:
We'll want to move the `type = nil` inside the loop like:

```ruby
type = nil
catch :no_literal do
 ...
```

Otherwise if there's a valid class parameter followed by an invalid one, the checker won't report an issue.